test_from_dict_bad_private_key breaks with cryptography 45.0.1

With cryptography>45 the following test breaks:

```pytb
google-auth> ________________________ test_from_dict_bad_private_key ________________________
google-auth> 
google-auth>     def test_from_dict_bad_private_key():
google-auth>         info = SERVICE_ACCOUNT_INFO.copy()
google-auth>         info["private_key"] = "garbage"
google-auth>     
google-auth>         with pytest.raises(ValueError) as excinfo:
google-auth>             _service_account_info.from_dict(info)
google-auth>     
google-auth> >       assert excinfo.match(r"key")
google-auth> E       AssertionError: Regex pattern did not match.
google-auth> E        Regex: 'key'
google-auth> E        Input: 'Unable to load PEM file. See https://cryptography.io/en/latest/faq/#why-can-t-i-import-my-pem-file for more details. MalformedFraming'
google-auth> 
google-auth> tests/test__service_account_info.py:56: AssertionError
```

https://cryptography.io/en/latest/faq/#why-can-t-i-import-my-pem-file

#### Environment details

  - OS: Linux (NixOS Unstable)
  - Python version: 3.13
  - pip version: n/a
  - `google-auth` version: 2.40.1

#### Steps to reproduce

  1. Update cryptography to 45.0.1
  2. Run the test suite

--- a/tests/test__service_account_info.py
+++ b/tests/test__service_account_info.py
@@ -53,7 +53,7 @@ def test_from_dict_bad_private_key():
     with pytest.raises(ValueError) as excinfo:
         _service_account_info.from_dict(info)
 
-    assert excinfo.match(r"key")
+    assert excinfo.match(r"(?i)(key|PEM)")
 
 
 def test_from_dict_bad_format():

